solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2400,15 +2402,15 @@ public void rejoinShardLeaderElection(SolrParams params) {
 
       try (SolrCore core = cc.getCore(coreName)) {
         Replica.Type replicaType = core.getCoreDescriptor().getCloudDescriptor().getReplicaType();
-        if (replicaType == Type.TLOG) {
+        if (replicaType.replicateFromLeader) {
           String leaderUrl =
               getLeader(
                   core.getCoreDescriptor().getCloudDescriptor(), cloudConfig.getLeaderVoteWait());
           if (!leaderUrl.equals(ourUrl)) {
```

Comment:
This is now checked for `PULL` replicas as well and is always `true`.
Should the `if` test the replica's `leaderEligible` flag in addition to `replicateFromLeader`?

Reply:
We are in `rejoinShardLeaderElection`, I assumed we can ever enter in this method for a PULL (or any non leader-eligible) replica, or can we?

Reply:
In the worst case, it doesn't hurt, since it is only used to verify whether a replica is the leader; PULL replica will not be, we would just have wasted time checking this. But I don't think we can ever end up in this method for a PULL replica.